tests/common/factories.py

```diff
@@ -104,6 +104,7 @@ class Meta:  # pylint: disable=no-init, old-style-class
     target_uri = factory.Faker('uri')
     text = factory.Faker('paragraph')
     userid = factory.Faker('user_name')
+    document = factory.SubFactory(Document)
```

Comment:
I was just trying this out and it only works if we can use the same session which might require some work.